examples/cli/src/main/java/io/unitycatalog/cli/ModelCli.java

```diff
@@ -0,0 +1,115 @@
+package io.unitycatalog.cli;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import io.unitycatalog.cli.utils.CliParams;
+import io.unitycatalog.cli.utils.CliUtils;
+import io.unitycatalog.client.ApiClient;
+import io.unitycatalog.client.ApiException;
+import io.unitycatalog.client.api.RegisteredModelsApi;
+import io.unitycatalog.client.model.CreateRegisteredModel;
+import io.unitycatalog.client.model.UpdateRegisteredModel;
+import java.util.List;
+import org.apache.commons.cli.CommandLine;
+import org.json.JSONObject;
+
+public class ModelCli {
+  private static final ObjectMapper objectMapper = CliUtils.getObjectMapper();
+  private static ObjectWriter objectWriter;
+
+  public static void handle(CommandLine cmd, ApiClient apiClient)
+      throws JsonProcessingException, ApiException {
+    RegisteredModelsApi registeredModelsApi = new RegisteredModelsApi(apiClient);
+    String[] subArgs = cmd.getArgs();
+    String subCommand = subArgs[1];
+    objectWriter = CliUtils.getObjectWriter(cmd);
+    JSONObject json = CliUtils.createJsonFromOptions(cmd);
+    String output = CliUtils.EMPTY;
+    switch (subCommand) {
+      case CliUtils.CREATE:
+        output = createRegisteredModel(registeredModelsApi, json);
+        break;
+      case CliUtils.LIST:
+        output = listRegisteredModels(registeredModelsApi, json);
+        break;
+      case CliUtils.GET:
+        output = getRegisteredModel(registeredModelsApi, json);
+        break;
+      case CliUtils.UPDATE:
+        output = updateRegisteredModel(registeredModelsApi, json);
+        break;
+      case CliUtils.DELETE:
+        output = deleteRegisteredModel(registeredModelsApi, json);
+        break;
+      default:
+        CliUtils.printEntityHelp(CliUtils.REGISTERED_MODEL);
+    }
+    CliUtils.postProcessAndPrintOutput(cmd, output, subCommand);
+  }
+
+  private static String createRegisteredModel(
+      RegisteredModelsApi registeredModelsApi, JSONObject json)
+      throws JsonProcessingException, ApiException {
+    CreateRegisteredModel createRegisteredModel;
```

Comment:
We can have more detailed CUJ's here involving model path and storing the actual model at the path (mostly a version) after obtaining temporary credentials for the same.

Reply:
So in the MLflow client, there is the abstraction of `log_model`.  This would go through the full process of:

- Create the registered model if it doesnt exist
- Create a new model version under that registered model
- Request credentials
- Upload artifacts of the model to the storage location
- Finalize the model version

We can have something similar to that in the cli but for now, and for easier testing by other devs, I'm adding all of these low level calls.  

The MLflow client is being worked on separately that will do all of those operations out of the box.